### PR TITLE
 Clarify requirements for Docker build 

### LINKED ISSUE #314: gradlew fails to build Docker image - gradle-wrapper.jar broken?
#### Software and hardware versions 
Aggregate master (git clone)
Ubuntu 16.04.5 LTS
```
java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
#### Problem description
* gradle/wrapper/gradle-wrapper.jar seems to be broken.
* Gradle's handling of `JAVA_HOME` seems broken, at least on Ubuntu, however that's not an Aggregate issue.

#### Steps to reproduce the problem
Follow the [Docker build](https://github.com/opendatakit/aggregate/blob/master/docs/build-and-run-a-docker-image.md).
* Setup local postgres database
* [Fix gradle's broken `JAVA_HOME`](https://stackoverflow.com/a/44168551/2813717) by running:
```
which javac >/dev/null 2>&1 || die "ERROR: no 'javac' command could be found in your PATH"
export JAVA_HOME=$(dirname $(dirname $(readlink -f $(which javac)  )))
```

* Run the gradle build step:
```
./gradlew clean dockerBuild -xtest -PwarMode=complete                                
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
```
However, [gradle/wrapper](https://github.com/opendatakit/aggregate/blob/master/gradle/wrapper) contains a gradle JAR. Naively running that JAR indicates it might be broken:
```
java -jar gradle/wrapper/gradle-wrapper.jar 
Error: Invalid or corrupt jarfile gradle/wrapper/gradle-wrapper.jar
```

#### Expected behavior
* Gradle builds the Docker image.

### Other information
Trying to rebuild gradle-wrapper.jar by running `gradle wrapper --stacktrace` fails with gradle v 1.2.
File wrapper.gradle contains:
```
task wrapper(type: Wrapper) {
   gradleVersion = '2.10' // version required
}
```
Updating gradle to 2.10, running `gradle wrapper` fails with:
```
Caused by: org.gradle.api.internal.MissingMethodException: Could not find method compileOnly() for arguments [{group=javax.servlet, name=javax.servlet-api, version=3.1.0}] on root project 'aggregate'.
```

### PR DESCRIPTION
**Pull request by florianm**
As encountered in issue #314, not following the correct dev environment setup in the master README will result in unexpected and confusing errors. The new additions are what I believe are the minimum working steps for a successful Docker build under Ubuntu 16.04. Windows and Mac environments/users may require additional steps/caveats.

### Closes 
#314

### What has been done to verify that this works as intended?
As documented in #314, I have successfully built a Docker image.

### Why is this the best possible solution? Were any other approaches considered?
Updating the docs should be sufficient to remind other builders of Docker images of the required steps for a successful build.

### Are there any risks to merging this code? If so, what are they?
No, it's only a change to docs.

### Do we need any specific form for testing your changes? If so, please attach one
No

### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
No, the ODK docs point to the Docker build README I'm proposing to update.

---
This change is [Reviewable](https://reviewable.io/reviews/opendatakit/aggregate/316)

### CODE REVIEW

File: docs/build-and-run-a-docker-image.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 - Install [Docker](https://www.docker.com)
 - Already have a PostgreSQL 9.6 database running
+- Follow the instructions of the main [README](https://github.com/opendatakit/aggregate), including:
```
Comment (ggalmazor) [inline docs/build-and-run-a-docker-image.md]: Do you think we could link to the [Getting the code](https://github.com/opendatakit/aggregate#getting-the-code) section directly?

```markdown
[Getting the code](https://github.com/opendatakit/aggregate#getting-the-code)
```

File: docs/build-and-run-a-docker-image.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
...
+  - Install Git LFS, as well as its git hooks, and enable the git hooks for Git LFS.
+  - Only then clone your fork of Aggregate - included .jar files only download correctly with Git LFS.
+  - Install IntelliJ IDEA and import project as instructed to let IDEA download, build, and configure sources.
```
Comment (ggalmazor) [inline docs/build-and-run-a-docker-image.md]: Here we could add a link to the specific place in the README that explains how to create the IntelliJ project.

```markdown
[Running the project - Import](https://github.com/opendatakit/aggregate#import)
```

**Comment by ggalmazor** [comment]
Thanks, @florianm!

I made some comments. Tell me what you think and we can move this PR forward.

File: docs/build-and-run-a-docker-image.md (hunk 1 above)
Comment (florianm) [reply docs/build-and-run-a-docker-image.md]: Good idea, have updated my PR!

**Comment by florianm** [comment]
not at all, thanks for the merge!

**Comment by ggalmazor** [comment]
It's been my pleasure :)

Any feedback on the Docker container will be welcome! We also have a Docker Compose setup that includes a container for the database.

Our next goal would be to build a Docker container that includes PostgreSQL and Tomcat, but we're facing some issues with that. It would be great to have some experienced user's help on that :wink:

**Pull request merged**